pennylane_lightning/lightning_kokkos/_state_vector.py

```diff
@@ -26,6 +52,433 @@ class LightningKokkosStateVector:  # pylint: disable=too-few-public-methods
         dtype: Datatypes for state-vector representation. Must be one of
             ``np.complex64`` or ``np.complex128``. Default is ``np.complex128``
         device_name(string): state vector device name. Options: ["lightning.kokkos"]
+        kokkos_args(InitializationSettings): binding for Kokkos::InitializationSettings
+            (threading parameters).
+        sync(bool): immediately sync with host-sv after applying operations
+
     """
 
-    pass  # pylint: disable=unnecessary-pass
+    def __init__(
+        self,
+        num_wires,
+        dtype=np.complex128,
+        device_name="lightning.kokkos",
+        kokkos_args=None,
+        sync=True,
+    ):  # pylint: disable=too-many-arguments
+        self._num_wires = num_wires
+        self._wires = Wires(range(num_wires))
+        self._dtype = dtype
+
+        self._kokkos_config = {}
+
+        if dtype not in [np.complex64, np.complex128]:  # pragma: no cover
+            raise TypeError(f"Unsupported complex type: {dtype}")
+
+        if device_name != "lightning.kokkos":
+            raise DeviceError(f'The device name "{device_name}" is not a valid option.')
+
+        self._device_name = device_name
+
+        # self._qubit_state = self._state_dtype()(self._num_wires)
+        if kokkos_args is None:
+            self._kokkos_state = self._state_dtype()(self.num_wires)
+        elif isinstance(kokkos_args, InitializationSettings):
+            self._kokkos_state = self._state_dtype()(self.num_wires, kokkos_args)
+        else:
+            type0 = type(InitializationSettings())
+            raise TypeError(
+                f"Argument kokkos_args must be of type {type0} but it is of {type(kokkos_args)}."
```

Comment:
Could we replace `{type0}` with `{type(InitializationSettings())}` here?

Reply:
Yes I will change it